lib/Core/Executor.cpp

```diff
@@ -1321,26 +1313,44 @@ Executor::toConstant(ExecutionState &state,
   if (ConstantExpr *CE = dyn_cast<ConstantExpr>(e))
     return CE;
 
-  ref<ConstantExpr> value;
-  bool success =
-      solver->getValue(state.constraints, e, value, state.queryMetaData);
-  assert(success && "FIXME: Unhandled solver failure");
-  (void) success;
+  ref<ConstantExpr> ce;
+  ref<Expr> value = nullptr;
+  if (auto found = seedMap.find(&state); found != seedMap.end())
+    value = getValueFromSeeds(found->second, e);
+  /* If no seed evaluation results in a constant, call the solver */
+  if (!value || !(ce = dyn_cast<ConstantExpr>(value))) {
+    bool success =
+        solver->getValue(state.constraints, e, ce, state.queryMetaData);
+    assert(success && "FIXME: Unhandled solver failure");
+    (void)success;
+  }
 
   std::string str;
   llvm::raw_string_ostream os(str);
   os << "silently concretizing (reason: " << reason << ") expression " << e
-     << " to value " << value << " (" << (*(state.pc)).info->file << ":"
+     << " to value " << ce << " (" << (*(state.pc)).info->file << ":"
      << (*(state.pc)).info->line << ")";
 
   if (ExternalCallWarnings == ExtCallWarnings::All)
     klee_warning("%s", os.str().c_str());
   else
     klee_warning_once(reason, "%s", os.str().c_str());
 
-  addConstraint(state, EqExpr::create(e, value));
-    
-  return value;
+  addConstraint(state, EqExpr::create(e, ce));
+
+  return ce;
+}
+
+ref<klee::Expr>
+Executor::getValueFromSeeds(std::vector<SeedInfo> &seeds, ref<Expr> e) {
+  ConstantExpr *CE = nullptr;
+  assert(!seeds.empty());
+  for (auto seed:seeds) {
+    auto value = seed.assignment.evaluate(e);
+    if ((CE = dyn_cast<ConstantExpr>(value)))
+      return CE;
+  }
+  return CE;
```

Comment:
```suggestion
  assert(!seeds.empty());
  for (auto const& seed : seeds) {
    auto value = seed.assignment.evaluate(e);
    if (isa<ConstantExpr>(value))
      return value;
  }
  return nullptr;
```

Captures  the intent much cleaner.

Note also the removal of the unnecessary (fairly expensive) copy of the loop variable